python/cudf/cudf/core/series.py

```diff
@@ -2745,14 +2744,27 @@ def quantile(
         0.75    3.25
         dtype: float64
         """
+        return_scalar = False
 
-        result = self._column.quantile(q, interpolation, exact)
+        if cudf.api.types.is_list_like(q) or cudf.utils.dtypes.is_column_like(
+            q
+        ):
+            np_array_q = cudf.core.column.as_column(q).values_host
+        elif is_scalar(q):
+            return_scalar = True
+            np_array_q = np.asarray([float(q)])
+        else:
+            raise TypeError(f"q must be a scalar or array-like, got {type(q)}")
```

Comment:
Would it also work to do
```suggestion
        return_scalar = is_scalar(q)
        try:
            np_array_q = cudf.core.column.as_column(q).values_host
        except WHATEVER_AS_COLUMN_RAISES:
            raise TypeError(f"q must be a scalar or array-like, got {type(q)}")
```

Then we wouldn't need to do `is_list_like` and `is_column_like`. We could also get rid of the `return_scalar = False` line, but I can't include that in this suggestion because of the intervening unmodified line.

Reply:
Hmm I guess the other question is whether we really need to do a device->host->device loop. What if we instead did
```
try:
    np_array_q = np.asarray(q)
except whatever:
    try:
        np_array_q = as_column(q).values_host
    except ...:
        ...
```
We should avoid the H2D2H copying if we can.

Reply:
Changed the logic to reduce such copies.